convert2rhel/checks.py

```diff
@@ -272,11 +275,19 @@ def check_custom_repos_are_valid():
 
 
 def ensure_compatibility_of_kmods():
-    """Ensure if the host kernel modules are compatible with RHEL."""
+    """Ensure that the host kernel modules are compatible with RHEL.
+
+    :raises SystemExit: Interrupts the conversion because some kernel modules are not supported in RHEL.
+    """
     host_kmods = get_loaded_kmods()
     rhel_supported_kmods = get_rhel_supported_kmods()
     unsupported_kmods = get_unsupported_kmods(host_kmods, rhel_supported_kmods)
-    if unsupported_kmods:
+
+    # Validate the best case first. If we don't have any unsupported_kmods, this means
+    # that everything is compatible and good to go.
+    if not unsupported_kmods:
+        logger.debug("Kernel modules are compatible.")
+    else:
```

Comment:
There's still one part of the RHELC-297 missing:

> Improve readability/maintainability of the `ensure_compatibility_of_kmods` function in checks.py as per https://github.com/oamg/convert2rhel/pull/193#discussion_r599881876

```python
def get_unsupported_kmods(host_kmods, rhel_supported_kmods):
    """Return a set of full paths to those installed kernel modules that are not available in RHEL repositories.
    ...
    """
    unsupported_kmods_subpaths = host_kmods - rhel_supported_kmods - set(system_info.kmods_to_ignore)
    unsupported_kmods_full_paths = ["/lib/modules/{kver}/{kmod}".format(kver=kernel_version, kmod=kmod)
                                    for kmod in unsupported_kmods_subpaths]
    return unsupported_kmods_full_paths
```

```python
def ensure_compatibility_of_kmods():
    ...
    if not unsupported_kmods:
        logger.debug("Kernel modules are compatible.")
    else:
        logger.critical(
            (
                "The following kernel modules are not supported in RHEL:\n{kmods}\n"
                "Make sure you have updated the kernel to the latest available version and rebooted the system.\n"
                "Prevent the unsupported modules from loading by following https://access.redhat.com/solutions/41278"
                " and run convert2rhel again to continue with the conversion."
            ).format(kmods="\n".join(unsupported_kmods), system=system_info.name)
        )
```

Reply:
Resolved by https://github.com/oamg/convert2rhel/pull/469/commits/5f52da81b3e23fb6a03cb4e5c2f4d251cb6b700f.